salt/loader.py

```diff
@@ -441,6 +441,7 @@ def roster(opts, whitelist=None):
         opts,
         tag='roster',
         whitelist=whitelist,
+        pack={'__runner__': runner(opts)})
```

Comment:
Doing this locks us into generating all runners every time we load rosters. IMHO, it's better to have this passed in so that we can reuse them in cases where runners have already been loaded by the caller.